tests/storage/migration.go

```diff
@@ -587,6 +591,132 @@ var _ = SIGDescribe("Volumes update with migration", decorators.RequiresTwoSched
+					libvmi.WithDataVolume(volName, dv.Name),
+					libvmi.WithCloudInitNoCloud(libvmifact.WithDummyCloudForFastBoot()),
+				)
+				createMigpolicyWithLimitedBandwidth(vmi)
+				vm := libvmi.NewVirtualMachine(vmi,
+					libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
+					libvmi.WithDataVolumeTemplate(dv),
+					libvmi.WithRunStrategy(virtv1.RunStrategyManual),
+				)
+				vm, err := virtClient.VirtualMachine(ns).Create(context.Background(), vm, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Starting the VM")
+				vm = libvmops.StartVirtualMachine(vm)
+				vmi = libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Running})
+				_, err = libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				return vm
+			}
+
+			It("when the copy of the destination volumes was successful", func() {
+				dv := createDV()
+				libstorage.CreateFSPVC(destPVC, ns, sizeWithOverhead, nil)
+				vm := createAndStartVM(dv)
+
+				By("Update volumes")
+				updateVMWithPVC(vm, volName, destPVC)
+
+				var migration *v1.VirtualMachineInstanceMigration
+				Eventually(func() v1.VirtualMachineInstanceMigrationPhase {
+					ls := labels.Set{
+						virtv1.VolumesUpdateMigration: vm.Name,
+					}
+					migList, err := virtClient.VirtualMachineInstanceMigration(ns).List(context.Background(),
+						metav1.ListOptions{
+							LabelSelector: ls.String(),
+						})
+					Expect(err).ToNot(HaveOccurred())
+					if len(migList.Items) < 1 {
+						return v1.MigrationPhaseUnset
+					}
+					Expect(migList.Items).To(HaveLen(1))
+					migration = &migList.Items[0]
+					return migration.Status.Phase
+				}).WithTimeout(time.Minute).WithPolling(time.Second).Should(Equal(v1.MigrationRunning))
+
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name,
+					metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Killing the source pod")
+				pod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+				err = virtClient.CoreV1().Pods(vmi.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{
+					GracePeriodSeconds: pointer.P(int64(0)),
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Waiting for the migration to be set as failed")
+				Eventually(func() v1.VirtualMachineInstanceMigrationPhase {
+					ls := labels.Set{
+						virtv1.VolumesUpdateMigration: vm.Name,
+					}
+					migList, err := virtClient.VirtualMachineInstanceMigration(ns).List(context.Background(),
+						metav1.ListOptions{
+							LabelSelector: ls.String(),
+						})
+					Expect(err).ToNot(HaveOccurred())
+					if len(migList.Items) < 1 {
+						return v1.MigrationPhaseUnset
+					}
+					Expect(migList.Items).To(HaveLen(1))
+					migration = &migList.Items[0]
+					return migration.Status.Phase
+				}).WithTimeout(time.Minute).WithPolling(time.Second).Should(Equal(virtv1.MigrationFailed))
```

Comment:
can this be shortened to https://github.com/kubevirt/kubevirt/blob/f3a46393dfb7dc17ef74807889cc81e34623b835/tests/migration/postcopy.go#L244-L245

Reply:
I cannot use `ThisMigration` since I don't know the name of the VMIM in advance. That's why I'm listing the migration with the volume migration label.
However, we don't need the migration variable

Reply:
Or I can use it here, but not in the previous step, when I wait the migration to be running